### PR TITLE
Temporal `Instant` migration cont. and related changes

### PR DESCRIPTION
**Pull request by nekevss**
This Pull Request related to ongoing work around #1804. This continues the work on migrating `Temporal` functionality in `boa_temporal`. 

It changes the following:

- Migrates `Instant` methods `add`, `subtract`, `round`, `until`, and `since` to `boa_temporal`
- Removes and/or migrates some functions

### CODE REVIEW

File: core/temporal/src/utils.rs [hunk 1]
```diff
@@ -104,6 +123,46 @@ pub(crate) fn round_number_to_increment(
...
+    // 1. Let quotient be x / increment.
+    let quotient = nanos / increment_nanos;
+    // 2. Let unsignedRoundingMode be GetUnsignedRoundingMode(roundingMode, false).
+    let unsigned_rounding_mode = rounding_mode.get_unsigned_round_mode(false);
+    // 3. Let r1 be the largest integer such that r1 ≤ quotient.
+    let r1 = quotient.floor();
+    // 4. Let r2 be the smallest integer such that r2 > quotient.
+    let r2 = quotient.ceil();
+    // 5. Let rounded be ApplyUnsignedRoundingMode(quotient, r1, r2, unsignedRoundingMode).
+    let rounded = apply_unsigned_rounding_mode(nanos, r1, r2, unsigned_rounding_mode);
```
Comment (raskad) [inline core/temporal/src/utils.rs:141]: Is the difference from the spec here intended? Passing `nanos` instead of `quotient`?

**Review by raskad — APPROVED** [review]
Looks good, I just have one question.

File: core/temporal/src/utils.rs (hunk 1 above)
Comment (nekevss) [reply core/temporal/src/utils.rs]: Nope, definitely not intended. 😅 Great catch!

**Review by HalidOdat — APPROVED** [review]
Looks good to me! :)

**Pull request merged**